base_classes/NXroot.nxdl.xml

```diff
@@ -46,15 +46,39 @@
     <attribute name="file_update_time" type="NX_DATE_TIME">
         <doc>Date and time of last file change at close</doc>
     </attribute>
+    <attribute name="NeXus_repository">
+        <doc>
+            A repository containing the application definitions
+            used for creating this file.
+            If the NeXus_version attribute contains a commit distance and hash
+            this should refer to this repository.
+        </doc>
+    </attribute>
     <attribute name="NeXus_version">
         <doc>
-            Version of NeXus API used in writing the file.
+            Version of NeXus definitions used in writing the file.
+            This may either be a date based version tag of the form `vYYYY.MM`
+            or a version tag with a commit distance and source control (e.g., git) hash of
+            the form `vYYYY.MM.post1.dev&lt;commit-distance&gt;.g&lt;git-hash&gt;`.
+            It may contain an additional `.dYYYYMMDD` timestamp appendix
+            for dirty repositories.
+            If the version contains a commit distance and hash the
+            NeXus_repository attribute should be written with the
+            repository url containing this version.
 
-            Only used when the NAPI has written the file.
+            Only used when the NAPI or pynxtools has written the file.
             Note that this is different from the version of the
             base class or application definition version number.
         </doc>
     </attribute>
+    <attribute name="partial">
+        <doc>
+             A list of concepts in an application definition this file describes.
+             This is for partially filling an application definition.
+             If this attribute is not present the application definition is assumed
+             to be valid, if not only the specified concepts/paths are assumed to be valid.
+        </doc>
+    </attribute>
```

Comment:
Remove `partial` in favor of using alternate solutions where smaller application definitions are used to validate subsets of files.

Reply:
removed

But, I will again state that I think the `partial` attribute that can be a very nice solution that can also be made rock solid if implemented correctly. Multiple smaller subsets of application definitions will not work if a vendor can only write a different subset of the application definition(s) depending on the settings of the instrument. In that case, one may need a big number of application definitions for each vendor individually.